### DISCUSSION TITLE
How to Start Storybook with a Special Config to Load Specified Stories

### DISCUSSION THREAD

**Discussion by g6th8b** [storybookjs/storybook#24652] [Help] [answered]
Hello GitHub Community,

I recently encountered an issue when trying to start Storybook with a custom configuration file to load only specified stories. The goal was to create a more streamlined development environment by configuring Storybook to focus on specific components during development.

Here's what I tried in my `package.json`:

```json
"scripts": {
  "storybook:test": "storybook -c ./test-config.js"
}
```

However, when I ran `yarn storybook:test`, I encountered the following error:

```
Error: No configuration files have been found in your configDir (../storybook\test-config.js).
Storybook needs "main.js" file, please add it (or pass a custom config dir flag to Storybook to tell where your main.js file is located at).
```

It seems like Storybook is expecting a `main.js` file in the specified config directory. This error got me wondering, is there a way to start Storybook with a special configuration file that tells it which stories to load? If so, what is the best way to achieve this?

I would appreciate any insights or guidance on this matter. Is there a parameter or method to set a special configuration for Storybook that defines which stories should be loaded?

Thank you for your help!

Best regards!

**Comment by ShaunEvening** [comment, marked as answer]
Hey @g6th8b 👋

The `-c` flag is for a custom config **directory** where it can find your `main.js`— not a path to a custom config file.

Copy your `main.js` and `preview.js` files from `.storybook` into a new directory like `.storybook-custom` with you custom story patterns. Once you've done that, you can update your `storybook:test` script to `storybook -c ./.storybook-custom`.

I hope that this helps 🙏 

- Shaun

**Reply by g6th8b** [reply]
Hello, Shaun! I have a question regarding content filtering. Is there an effective method to implement keyword-based whitelisting and blacklisting for stories or content? I want to be able to search through containers of content, and use regular expressions to either selectively load or exclude stories based on specific keywords. How can I achieve this?

Thank you very much!

**Reply by ShaunEvening** [reply]
Hey @g6th8b 👋 

I'm not sure that that is currently possible. We're hoping to implement a tagging system to let uses filter their stories and docs based on given tags.
